File: FacadeCore/IDepartmentService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using BOCore;

namespace FacadeCore
{
    public interface IDepartmentService
    {
        GetResponse<DepartmentBO> GetDepartments();
        GetResponse<IdNameBO> GetDepartmentsForSelect();
        GetResponse<DepartmentBO> GetDepartmentById(int id);
        GetResponse<DepartmentBO> GetDepartmentByIds(List<int> IdList);
        Response InsertUpdate(DepartmentBO bo);
        Response Delete(List<int> ids);
        Response Delete(List<DepartmentBO> bos);
    }
}

File: Facade/IDepartmentService.vb
Imports BO

Public Interface IDepartmentService
    Function GetDepartments() As GetResponse(Of DepartmentBO)
    Function GetDepartmentsForSelect() As GetResponse(Of IdNameBO)
    Function GetDepartmentById(id As Integer) As GetResponse(Of DepartmentBO)
    Function GetDepartmentByIds(IdList As List(Of Integer)) As GetResponse(Of DepartmentBO)
    Function InsertUpdate(bo As DepartmentBO) As Response
    Function Delete(ids As List(Of Integer)) As Response
    Function Delete(bos As List(Of DepartmentBO)) As Response
End Interface
